### DISCUSSION TITLE
Asset hot reload

### DISCUSSION THREAD

**Discussion by ma-hnln** [stride3d/stride#2388] [Q&A]
Hi,

is there a way to hot reload assets into a running game?

While digging through the code I already stumbled across `Stride.Editor.EditorGame.ContentLoader.EditorContentLoader`  which seems to do hot reloading of resources/assets for the in editor scenes. Is there a similar thing for games running outside of the editor?

Generally, I would like to explore the code base a bit more and experiment with it. Is there any design document or similar documentation which would help in understanding asset related systems and how they work/what they expect when the game is running? [This](https://doc.stride3d.net/latest/en/contributors/engine/architecture/build-details.html) just mentions that the asset compiler runs as part of a games' build process. 

Thanks!

**Comment by Doprez** [comment]
The closest things I know of at a high level are the following:
[dependancy graphs thanks to @VaclavElias](https://stride-doc-staging.azurewebsites.net/latest/en/contributors/engine/architecture/dependency-graph.html)
[Some things I follow but mostly relate to UI not asset loading sadly](https://github.com/Doprez/Awesome-Stride?tab=readme-ov-file#engine-documentation)

I would love more documentation on the Asset pipeline as its a bit of a black box right now to me but there are a few projects that kind of help explaining it through hands on code examples. These are after the `EditorContentLoader` though so they are a bit lower level in the chain.
[scriptable objects test example](https://github.com/manio143/StrideCustomAsset)
[load gltf files at runtime](https://github.com/ykafia/StrideGltf/blob/master/GltfImport/LoadGLTF.cs)
[load obj files at runtime](https://github.com/jeske/LoadObjTest)

**Reply by VaclavElias** [reply]
I would like to create dependency graphs also for the individual parts of Stride but that is in my low priority backlog.
